### DISCUSSION TITLE
SSL?

### DISCUSSION THREAD

**Discussion by huperaisan** [crocodilestick/Calibre-Web-Automated#105] [General]
Just wondering how/whether you guys are passing or obtaining an SSL certificate for the docker-compose.

**Comment by jmarmstrong1207** [comment]
For me and many others, we use a reverse proxy for SSL. I use NGINX Proxy Manager as a separate docker container, and it features a GUI to easily set up SSL

**Reply by huperaisan** [reply]
I’ve been trying to figure out how that works, because I didn’t know how different Docker containers interact with each other (like how nginx works with CWA). Do you have a link handy that explains how to set this up?

**Reply by jmarmstrong1207** [reply]
> I’ve been trying to figure out how that works, because I didn’t know how different Docker containers interact with each other (like how nginx works with CWA). Do you have a link handy that explains how to set this up?

The reverse proxies communicate with the server apps via their ports. All they do is redirect traffic to those ports based on the domain url the user enters, e.g calibre.example.com redirects to port 8083. I don’t have a setup guide specifically, but any guide online should do. All u do is add a new entry in the admin UI for redirecting a domain to the internal ip and port of the web server. There’s a tab in the creation menu to enable SSL and it’ll set it up with let’s encrypt

**Reply by huperaisan** [reply]
I'll check this out! I had been trying to figure out how to do this because I'd installed Docker on an instance that was set up with HestiaCP, but I've decided to just tear everything up and start over on a clean instance.
